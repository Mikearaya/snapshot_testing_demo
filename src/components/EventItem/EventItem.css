.event-image {
  width: 15em;
  height: 15em;
  border-color: chartreuse;
}

.container {
  text-align: center !important;
}

.start-date {
  display: block;
  font-weight: 300;
  text-align: left;
}

ul {
  list-style: none;
}

ul li {
  display: inline;
  float: left;
  width: 32%;
}
