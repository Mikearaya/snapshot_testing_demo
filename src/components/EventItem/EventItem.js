import './EventItem.css';

import React, { Component } from 'react';

import PropTypes from 'prop-types';

class EventItem extends Component {
  render() {
    return (
      <div className='container'>
        <p>{this.props.name}</p>
        <span className='start-date'> {this.props.startDate}</span>
        <p>{this.props.description}</p>
      </div>
    );
  }
}

EventItem.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired
};

EventItem.defaultProps = {
  name: '',
  description: '',
  startDate: ''
};

export default EventItem;
