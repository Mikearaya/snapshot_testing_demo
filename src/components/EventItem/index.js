import EventItem from './EventItem';

export default EventItem;
