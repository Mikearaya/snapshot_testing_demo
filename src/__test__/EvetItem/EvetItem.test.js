import EventItem from '../../components/EventItem';
import React from 'react';
import renderer from 'react-test-renderer';

describe('EventList Component', () => {
  test('should render correctly', () => {
    const component = renderer.create(<EventItem />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
