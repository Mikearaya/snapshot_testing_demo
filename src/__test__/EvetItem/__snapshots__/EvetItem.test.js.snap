// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList Component should render correctly 1`] = `
<div
  className="container"
>
  <p>
    
  </p>
  <span
    className="start-date"
  >
     
    
  </span>
  <p>
    
  </p>
</div>
`;
